sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1735,8 +1735,22 @@ case class AnsiCast(child: Expression, dataType: DataType, timeZoneId: Option[St
 /**
  * Cast the child expression to the target data type, but will throw error if the cast might
  * truncate, e.g. long -> int, timestamp -> data.
+ *
+ * Note that UpCast will be eliminated if the child's dataType is already DecimalType and
```

Comment:
We can simplify the doc:
```
Note: `target` is `AbstractDataType`, so that we can put `object DecimalType`, which means we accept
`DecimalType` with any valid precision/scale.
```